### PR TITLE
Decrypted db objects are transformed.

### PR DESCRIPTION
**Pull request by DanielChicot**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -63,24 +63,24 @@ successful processing step.
...
 * `SourceRecordProcessor` - simply reads and returns the value of the kafka
   consumer record.
+* `ValidationProcessor` - filters out messages that are not valid against the message schema.
 * `JsonProcessor` - parse the source record value as a Json object.
 * `ExtractionProcessor` - extracts the encryption metadata from the Json
   object (the encrypted key, the encrypting key id, and the initialisation
       vector generated when encrypting the payload).
 * `DatakeyProcessor` - calls DKS to get the encrypted datakey decrypted.
 * `DecryptionProcessor` - decrypts the `dbObject`
+* `TransformationProcesser` - transforms the decrypted dbObject in line with
```
Comment (tea-short) [inline README.md:73]: typo in TransformationProcessOr :)

File: src/main/kotlin/ucfs/claimant/consumer/properties/AwsProperties.kt [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package ucfs.claimant.consumer.properties
+
+import org.springframework.boot.context.properties.ConfigurationProperties
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+
+@Configuration
+@ConfigurationProperties(prefix = "aws")
+data class AwsProperties(var saltParameterName: String = "/ucfs/claimant-api/nino/salt",
+                         var cmkAlias: String = "alias/ucfs_etl_cmk") {
```
Comment (dan-hill2802) [inline src/main/kotlin/ucfs/claimant/consumer/properties/AwsProperties.kt:10]: I don't think the parameter name or CMK alias should be here, I don't think they should have a default value

File: README.md [hunk 3]
```diff
@@ -94,24 +94,42 @@ The application can be configured with spring properties (specified on the
...
-| kafka.topicRegex             | KAFKA_TOPIC_REGEX               | Topics matching this regex will be subscribed to |
-| kafka.useSsl                 | KAFKA_USE_SSL                   | Whether to enable a mutually authenticated connection |
-| security.keyPassword         | SECURITY_KEY_PASSWORD           | Private key password |
-| security.keystore            | SECURITY_KEYSTORE               | Path to keystore containing app certificates |
-| security.keystorePassword    | SECURITY_KEYSTORE_PASSWORD      | Keystore password |
-| security.truststore          | SECURITY_TRUSTSTORE             | Path to keystore containing trusted certificates, needs to allow access from DKS and the kafka broker |
-| security.truststorePassword  | SECURITY_TRUSTSTORE_PASSWORD    | Truststore password |
+| Property                        | Environment variable              | Purpose | Default | Needs override |
+|---------------------------------|-----------------------------------|---------|---------|----------------|
+| aws.cmkAlias                    | AWS_CMK_ALIAS                     | The alias of the master ky in KMS | "alias/ucfs_etl_cmk" | |
```
[the remark this answers is not in the document]
Comment (DanielChicot) [reply README.md:99]: I like this feature :+1:

File: src/main/kotlin/ucfs/claimant/consumer/properties/AwsProperties.kt (hunk 2 above)
Comment (DanielChicot) [reply src/main/kotlin/ucfs/claimant/consumer/properties/AwsProperties.kt:10]: Removed the defaults and updated the docs.

File: README.md [hunk 4]
```diff
@@ -94,24 +94,42 @@ The application can be configured with spring properties (specified on the
...
-| kafka.topicRegex             | KAFKA_TOPIC_REGEX               | Topics matching this regex will be subscribed to |
-| kafka.useSsl                 | KAFKA_USE_SSL                   | Whether to enable a mutually authenticated connection |
-| security.keyPassword         | SECURITY_KEY_PASSWORD           | Private key password |
-| security.keystore            | SECURITY_KEYSTORE               | Path to keystore containing app certificates |
-| security.keystorePassword    | SECURITY_KEYSTORE_PASSWORD      | Keystore password |
-| security.truststore          | SECURITY_TRUSTSTORE             | Path to keystore containing trusted certificates, needs to allow access from DKS and the kafka broker |
-| security.truststorePassword  | SECURITY_TRUSTSTORE_PASSWORD    | Truststore password |
+| Property                        | Environment variable              | Purpose | Default | Needs override |
+|---------------------------------|-----------------------------------|---------|---------|----------------|
+| aws.cmkAlias                    | AWS_CMK_ALIAS                     | The alias of the master ky in KMS |  | Yes |
```
Comment (dan-hill2802) [inline README.md:99]:
```suggestion
| aws.cmkAlias                    | AWS_CMK_ALIAS                     | The alias of the master key in KMS |  | Yes |
```

**Review by dan-hill2802 — APPROVED** [review]

**Pull request merged**